packages/data-context/src/sources/GitDataSource.ts

```diff
@@ -302,9 +302,13 @@ export class GitDataSource {
 
     debug('executing command `%s`:', cmd)
 
-    const result = await execa(cmd, { shell: process.env.SHELL || '/bin/bash' })
+    const result = await execa(cmd, { shell: process.env.SHELL || '/bin/bash', cwd: this.#gitBaseDir })
```

Comment:
Can we add a test covering this change in the `GitDataSource.spec.ts`?

Reply:
I'm not sure if there's value in writing a test for this. I was able to find the problem by opening the cypress app in from within the cypress folder via `yarn cypress:open` and then navigating to list specs for a project. The issue was due to the fact that the cypress project folder was the default working directory since the spawned process would inherit that from the main process, which may be an OS-level or NodeJS behavior.
If you think we still need a test, then would this be a negative test by opening cypress in global mode?

Reply:
I meant in the unit tests file it should be pretty simple to simulate the behavior you're describing, just a `process.chdir` or something to make a failing test that this change would make green, just a general practice to guards against it being lost in a future refactor: https://github.com/cypress-io/cypress/blob/1ed93ffebd446532e3df1b50913a60874b4c6ef1/packages/data-context/test/unit/sources/GitDataSource.spec.ts#L36-L47